### DISCUSSION TITLE
Support for OPC UA 1.04

### DISCUSSION THREAD

**Discussion by br-iosb** [eclipse-milo/milo#811] [General]
Hello,

as far as I understand, Milo does not yet support OPC UA v1.04 and will not before Milo reaches version 1.0.
Therefore I have the following questions:
What is still missing until 1.0 is reached?
How extensive will an update to OPC UA v1.04 be?

Greetings
Bastian

**Comment by kevinherron** [comment]
Not much is missing before 1.0 can be released, it will probably happen soon. I sent the following email to the mailing list recently:

> All,
> 
> The release of Milo 0.6.0 is imminent.
> 
> This is probably the last "major" release before version 1.0. Milo has been functionally stable for a long time now and I don't foresee any further breaking API changes. The demo server built on the server SDK passes the full OPC UA Compliance Test Tool "Standard UA Server" suite. Any interim 0.6.x releases will be bug fixes as necessary unless somebody out there has a compelling reason otherwise.
> 
> When 1.0 is released all currently deprecated methods will be removed. Now is the time to migrate your code.
> 
> Development in the 1.0 release series is likely to be bug fixes only. I will soon be branching "dev/2.0" (possibly before 1.0 is even released) to begin work on implementing OPC UA 1.04/1.05.

OPC UA 1.04 will be a lot of work and probably break some APIs again. It will also be moving the requirement for JDK 8 to JDK 11.
